infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -175,8 +175,8 @@ private void setDataConsistentWithRetry(boolean isDataConsistent) {
         try {
             IRetry.build(IntervalRetry.class, () -> {
                 try {
-                    dataConsistent.set(isDataConsistent);
                     logReplicationMetadataManager.setDataConsistentOnStandby(isDataConsistent);
+                    dataConsistent.set(isDataConsistent);
```

Comment:
The in-memory copy must be updated only after it is persisted in Corfu.  Otherwise there can be an inconsistency between the 2 copies if the write transaction fails.